--- linux-6.12.62.orig/drivers/char/lrng/lrng_drng_chacha20.h	2025-12-12 20:56:13.629657786 +0100
+++ linux-6.12.62/drivers/char/lrng/lrng_drng_chacha20.h	2025-12-12 21:06:46.476553370 +0100
@@ -28,7 +28,7 @@ struct chacha20_state {
 
 static inline void lrng_cc20_init_rfc7539(struct chacha20_block *chacha20)
 {
-	chacha_init_consts((struct chacha_state *)chacha20->constants);
+	chacha_init_consts(chacha20->constants);
 }
 
 #define LRNG_CC20_INIT_RFC7539(x) \
--- linux-6.12.62.orig/drivers/char/lrng/lrng_drng_chacha20.c	2025-12-12 20:56:13.629468142 +0100
+++ linux-6.12.62/drivers/char/lrng/lrng_drng_chacha20.c	2025-12-12 21:11:46.052573178 +0100
@@ -37,7 +37,7 @@ static void lrng_chacha20_update(struct
 	BUILD_BUG_ON(CHACHA_BLOCK_SIZE != 2 * CHACHA_KEY_SIZE);
 
 	if (used_words > CHACHA_KEY_SIZE_WORDS) {
-		chacha20_block((struct chacha_state *)chacha20, (u8 *)tmp);
+		chacha20_block(&chacha20->constants[0], (u8 *)tmp);
 		for (i = 0; i < CHACHA_KEY_SIZE_WORDS; i++)
 			chacha20->key.u[i] ^= le32_to_cpu(tmp[i]);
 		memzero_explicit(tmp, sizeof(tmp));
@@ -107,13 +107,13 @@ static int lrng_cc20_drng_generate_helpe
 	int zeroize_buf = 0;
 
 	while (outbuflen >= CHACHA_BLOCK_SIZE) {
-		chacha20_block((struct chacha_state *)chacha20, outbuf);
+		chacha20_block(&chacha20->constants[0], outbuf);
 		outbuf += CHACHA_BLOCK_SIZE;
 		outbuflen -= CHACHA_BLOCK_SIZE;
 	}
 
 	if (outbuflen) {
-		chacha20_block((struct chacha_state *)chacha20, (u8 *)aligned_buf);
+		chacha20_block(&chacha20->constants[0], (u8 *)aligned_buf);
 		memcpy(outbuf, aligned_buf, outbuflen);
 		used = ((outbuflen + sizeof(aligned_buf[0]) - 1) /
 			sizeof(aligned_buf[0]));
