--- linux-6.12.62.orig/kernel/sched/core.c	2025-12-12 20:52:36.608404255 +0100
+++ linux-6.12.62/kernel/sched/core.c	2025-12-12 20:53:11.057504849 +0100
@@ -7,8 +7,6 @@
  *  Copyright (C) 1991-2002  Linus Torvalds
  *  Copyright (C) 1998-2024  Ingo Molnar, Red Hat
  */
-#define INSTANTIATE_EXPORTED_MIGRATE_DISABLE
-#include <linux/sched.h>
 #include <linux/lrng.h>
 #include <linux/highmem.h>
 #include <linux/hrtimer_api.h>
