--- linux-6.12.62.orig/kernel/sched/core.c	2025-12-12 18:37:22.000000000 +0100
+++ linux-6.12.62/kernel/sched/core.c	2025-12-12 20:52:21.162357901 +0100
@@ -7,6 +7,8 @@
  *  Copyright (C) 1991-2002  Linus Torvalds
  *  Copyright (C) 1998-2024  Ingo Molnar, Red Hat
  */
+#define INSTANTIATE_EXPORTED_MIGRATE_DISABLE
+#include <linux/sched.h>
 #include <linux/highmem.h>
 #include <linux/hrtimer_api.h>
 #include <linux/ktime_api.h>
