--- linux-6.12.62.orig/drivers/char/lrng/lrng_sysctl.c	2025-12-12 20:56:13.719246227 +0100
+++ linux-6.12.62/drivers/char/lrng/lrng_sysctl.c	2025-12-12 21:08:03.289307192 +0100
@@ -86,7 +86,7 @@ void lrng_sysctl_update_max_write_thresh
 	mb();
 }
 
-static const struct ctl_table random_table[] = {
+static struct ctl_table random_table[] = {
 	{
 		.procname	= "poolsize",
 		.maxlen		= sizeof(int),
