--- a/ecm_classifier_default.c
+++ b/ecm_classifier_default.c
@@ -655,7 +655,7 @@ static int ecm_classifier_default_state_
  * ecm_classifier_default_enable_handler()
  * 	Default classifier enable/disable sysctl node handler
  */
-static int ecm_classifier_default_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_default_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -699,7 +699,7 @@ static int ecm_classifier_default_enable
  * ecm_classifier_default_accel_mode_handler()
  * 	Proc handler to change acceleration mode
  */
-static int ecm_classifier_default_accel_mode_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_default_accel_mode_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -754,7 +754,7 @@ static int ecm_classifier_default_accel_
  * 	Proc handler to handle slow path packets allowed
  * 	before acceleration
  */
-static int ecm_classifier_accel_delay_pkts_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_accel_delay_pkts_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_dscp.c
+++ b/ecm_classifier_dscp.c
@@ -896,7 +896,7 @@ static int ecm_classifier_dscp_state_get
  * ecm_classifier_dscp_enable_handler()
  * 	Proc handler to enable or disable DSCP classifier
  */
-static int ecm_classifier_dscp_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_dscp_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_emesh.c
+++ b/ecm_classifier_emesh.c
@@ -4070,7 +4070,7 @@ defunct_by_priority:
  * ecm_classifier_emesh_enable_handler()
  * 	Proc handler to enable or disable emesh classifier
  */
-static int ecm_classifier_emesh_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_emesh_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -4117,7 +4117,7 @@ static int ecm_classifier_emesh_enable_h
  * ecm_classifier_emesh_latency_config_enable_handler()
  * 	Proc handler to enable or disable mesh latency
  */
-static int ecm_classifier_emesh_latency_config_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_emesh_latency_config_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -4164,7 +4164,7 @@ static int ecm_classifier_emesh_latency_
  * ecm_classifier_sawf_enable_handler()
  * 	Proc handler to enable or disable SAWF
  */
-static int ecm_classifier_sawf_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_sawf_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -4210,7 +4210,7 @@ static int ecm_classifier_sawf_enable_ha
  * ecm_classifier_sawf_cake_enable_handler()
  * 	Proc handler to enable or disable CAKE Qdisc flag
  */
-static int ecm_classifier_sawf_cake_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_sawf_cake_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -4254,7 +4254,7 @@ static int ecm_classifier_sawf_cake_enab
  * ecm_classifier_3link_mlo_enable_handler()
  * 	Proc handler to enable or disable 3 link mlo
  */
-static int ecm_classifier_3link_mlo_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_3link_mlo_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -4298,7 +4298,7 @@ static int ecm_classifier_3link_mlo_enab
  * ecm_classifier_sawf_emesh_udp_ipsec_port_handler()
  * 	Proc handler to update UDP ipsec port
  */
-static int ecm_classifier_sawf_emesh_udp_ipsec_port_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_sawf_emesh_udp_ipsec_port_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_mark.c
+++ b/ecm_classifier_mark.c
@@ -629,7 +629,7 @@ static int ecm_classifier_mark_state_get
  * ecm_classifier_mark_enable_handler()
  * 	Proc handler to enable or disable MARK classifier
  */
-static int ecm_classifier_mark_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_mark_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_mscs.c
+++ b/ecm_classifier_mscs.c
@@ -1225,7 +1225,7 @@ static int ecm_classifier_mscs_scs_set_u
  * ecm_classifier_mscs_enable_handler()
  *  Proc handler to enable or disable mscs classifier
  */
-static int ecm_classifier_mscs_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_mscs_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1268,7 +1268,7 @@ static int ecm_classifier_mscs_enable_ha
  * ecm_classifier_scs_enable_handler()
  * 	Proc handler to enable or disable scs classifier
  */
-static int ecm_classifier_scs_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_scs_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1311,7 +1311,7 @@ static int ecm_classifier_scs_enable_han
  * ecm_classifier_mscs_scs_multi_ap_enable_handler()
  * 	Proc handler to enable/disable Multi AP for MSCS classifier
  */
-static int ecm_classifier_mscs_scs_multi_ap_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_mscs_scs_multi_ap_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1355,7 +1355,7 @@ static int ecm_classifier_mscs_scs_multi
  * ecm_classifier_mscs_scs_udp_ipsec_port_handler()
  * 	Proc handler to update UDP IPsec port
  */
-static int ecm_classifier_mscs_scs_udp_ipsec_port_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_mscs_scs_udp_ipsec_port_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_ovs.c
+++ b/ecm_classifier_ovs.c
@@ -2184,7 +2184,7 @@ done:
  * ecm_classifier_ovs_enable_handler()
  * 	Proc handler to enable or disable OVS classifier
  */
-static int ecm_classifier_ovs_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_ovs_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_pcc.c
+++ b/ecm_classifier_pcc.c
@@ -1320,7 +1320,7 @@ static int ecm_classifier_pcc_state_get(
  * ecm_classifier_pcc_enable_handler()
  * 	Proc handler to enable or disable PCC classifier
  */
-static int ecm_classifier_pcc_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_pcc_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_classifier_wifi.c
+++ b/ecm_classifier_wifi.c
@@ -675,7 +675,7 @@ static void ecm_classifier_wifi_should_k
  * ecm_classifier_wifi_enable_handler
  * 	Proc handler to enable or disable WIFI classifier
  */
-static int ecm_classifier_wifi_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_wifi_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_conntrack_notifier.c
+++ b/ecm_conntrack_notifier.c
@@ -383,7 +383,7 @@ static int ecm_conntrack_event(unsigned
  * ecm_conntrack_notifier_stop_handler()
  * 	Proc handler to stop conntrack notifier
  */
-static int ecm_conntrack_notifier_stop_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_conntrack_notifier_stop_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_db/ecm_db.c
+++ b/ecm_db/ecm_db.c
@@ -476,7 +476,7 @@ static struct notifier_block ecm_db_ip6r
  * ecm_db_defunct_all_handler()
  * 	Proc handler to defunct ecm db
  */
-static int ecm_db_defunct_all_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_db_defunct_all_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -525,7 +525,7 @@ static int ecm_db_defunct_all_handler(st
  * ecm_db_ipv4_route_handler()
  * 	Proc handler to disable IPV4 route table events
  */
-static int ecm_db_ipv4_route_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_db_ipv4_route_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -586,7 +586,7 @@ static int ecm_db_ipv4_route_handler(str
  * 	Proc handler to disable IPV6 route table events
  */
 #ifdef ECM_IPV6_ENABLE
-static int ecm_db_ipv6_route_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_db_ipv6_route_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_db/ecm_db_connection.c
+++ b/ecm_db/ecm_db_connection.c
@@ -4996,7 +4996,7 @@ next_v0:
  * ecm_db_connection_defunct_by_5tuple_mask_handler()
  * 	Proc handler to defunct the ecm rules by giving 5 tuple mask
  */
-static int ecm_db_connection_defunct_by_5tuple_mask_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_db_connection_defunct_by_5tuple_mask_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
@@ -5218,7 +5218,7 @@ bool ecm_db_connection_defunct_5tuple_bu
  * ecm_db_connection_defunct_by_5tuple_handler()
  * 	Proc handler to defunct the ecm rules by giving 5 tuple
  */
-static int ecm_db_connection_defunct_by_5tuple_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_db_connection_defunct_by_5tuple_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -9003,7 +9003,7 @@ static struct notifier_block ecm_interfa
  * ecm_interface_igs_enabled_handler()
  *	IGS enabled check sysctl node handler.
  */
-static int ecm_interface_igs_enabled_handler(struct ctl_table *ctl, int write, void __user *buffer,
+static int ecm_interface_igs_enabled_handler(const struct ctl_table *ctl, int write, void __user *buffer,
 		 size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -9044,7 +9044,7 @@ static int ecm_interface_igs_enabled_han
  * ecm_interface_src_check_handler()
  *	Source interface check sysctl node handler.
  */
-static int ecm_interface_src_check_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_src_check_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_value;
@@ -9085,7 +9085,7 @@ static int ecm_interface_src_check_handl
  * ecm_interface_src_check_no_flush_handler()
  *	Source interface check no flush sysctl node handler.
  */
-static int ecm_interface_src_check_no_flush_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_src_check_no_flush_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
@@ -9109,7 +9109,7 @@ static int ecm_interface_src_check_no_fl
  * ecm_interface_mwan3_enable_handler()
  *	 mwan3 enable check sysctl node handler.
  */
-static int ecm_interface_mwan3_enable_handler(struct ctl_table *ctl, int write, void __user *buffer,
+static int ecm_interface_mwan3_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer,
 		 size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -9335,7 +9335,7 @@ static int ecm_interface_accel_denied_ha
  * ecm_interface_accel_denied_list_handler()
  * 	Proc handler function for denied interface read/write operation
  */
-static int ecm_interface_accel_denied_list_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_accel_denied_list_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -9522,7 +9522,7 @@ static int ecm_interface_defunct_by_ifac
  * ecm_interface_defunct_by_mac_addr_handler()
  * 	Proc handler function for defunct the ecm rules by mac address
  */
-static int ecm_interface_defunct_by_mac_address_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_defunct_by_mac_address_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * To mark a MAC address as defunct:
@@ -9537,7 +9537,7 @@ static int ecm_interface_defunct_by_mac_
  * ecm_interface_defunct_by_iface_handler()
  * 	Proc handler function for defunct the ecm rules by iface
  */
-static int ecm_interface_defunct_by_iface_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_interface_defunct_by_iface_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * To mark a iface name as defunct:
--- a/ecm_sdx/ecm_sdx.c
+++ b/ecm_sdx/ecm_sdx.c
@@ -73,7 +73,7 @@ static void ecm_sdx_host_removed(void *a
  * ecm_sdx_stats_handler()
  *	Proc handler function for stats read/write operation.
  */
-static int ecm_sdx_stats_handler(struct ctl_table *ctl, int write, void *buffer,
+static int ecm_sdx_stats_handler(const struct ctl_table *ctl, int write, void *buffer,
 				size_t *lenp, loff_t *ppos)
 {
 	/*
@@ -101,7 +101,7 @@ static int ecm_sdx_stats_handler(struct
  * ecm_sdx_per_client_stats_handler()
  *	Proc handler function for packet_stats read/write operation.
  */
-static int ecm_sdx_per_client_stats_handler(struct ctl_table *ctl, int write, void *buffer,
+static int ecm_sdx_per_client_stats_handler(const struct ctl_table *ctl, int write, void *buffer,
 					size_t *lenp, loff_t *ppos)
 {
 	/*
@@ -141,7 +141,7 @@ static int ecm_sdx_per_client_stats_hand
  * ecm_sdx_iface_type_handler()
  *	Proc handler function for packet_stats_interface read/write operation.
  */
-static int ecm_sdx_iface_type_handler(struct ctl_table *ctl, int write, void *buffer,
+static int ecm_sdx_iface_type_handler(const struct ctl_table *ctl, int write, void *buffer,
 					size_t *lenp, loff_t *ppos)
 {
 	/*
@@ -167,7 +167,7 @@ static int ecm_sdx_iface_type_handler(st
  * ecm_sdx_lan_prefix_handler()
  *	Proc handler function for lan_prefixes read/write operation.
  */
-static int ecm_sdx_lan_prefix_handler(struct ctl_table *ctl, int write, void *buffer,
+static int ecm_sdx_lan_prefix_handler(const struct ctl_table *ctl, int write, void *buffer,
 					size_t *lenp, loff_t *ppos)
 {
 	/*
--- a/ecm_state.c
+++ b/ecm_state.c
@@ -886,7 +886,7 @@ static struct file_operations ecm_state_
  * ecm_state_file_output_mask_handler()
  * 	Proc handler to select different output flags for ecm dump
  */
-static int ecm_state_file_output_mask_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_state_file_output_mask_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/ecm_tracker_udp.c
+++ b/ecm_tracker_udp.c
@@ -1160,7 +1160,7 @@ static int ecm_tracker_rtp_payload_types
  * ecm_tracker_udp_rtp_payload_types_handler()
  *	Proc handler function for RTP payload types read/write operation.
  */
-static int ecm_tracker_udp_rtp_payload_types_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_tracker_udp_rtp_payload_types_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1180,7 +1180,7 @@ static int ecm_tracker_udp_rtp_payload_t
  * ecm_tracker_udp_clf_enabled_handler()
  *	Sysctl to enable/disable UDP classifier flag.
  */
-int ecm_tracker_udp_clf_enabled_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_tracker_udp_clf_enabled_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
--- a/examples/ecm_ae_select.c
+++ b/examples/ecm_ae_select.c
@@ -646,7 +646,7 @@ static int ecm_ae_select_test_rule_read_
  * ecm_ae_select_test_rule_handler()
  * 	Sysctl handler for AE select test rule ops
  */
-static int ecm_ae_select_test_rule_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ae_select_test_rule_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
--- a/examples/ecm_pcc_test.c
+++ b/examples/ecm_pcc_test.c
@@ -1447,7 +1447,7 @@ static int ecm_pcc_test_rule_read_handle
  * ecm_pcc_test_rule_handler()
  * 	Sysctl handler for PCC test rule operations
  */
-static int ecm_pcc_test_rule_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_pcc_test_rule_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
@@ -1483,7 +1483,7 @@ static int ecm_pcc_test_rule_handler(str
  * ecm_pcc_test_unregister_handler()
  * 	Sysctl handler to unregister PCC test classifer
  */
-static int ecm_pcc_test_unregister_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_pcc_test_unregister_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * return if the operation is read(cat /proc/sys/net/ecm_pcc_test/unregister)
@@ -1507,7 +1507,7 @@ static int ecm_pcc_test_unregister_handl
  * ecm_pcc_test_pcc_mirror_en_handler()
  * 	Sysctl handler for pcc_mirror_en with bounds checking
  */
-static int ecm_pcc_test_pcc_mirror_en_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_pcc_test_pcc_mirror_en_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_val;
--- a/examples/ecm_sdx_pcc.c
+++ b/examples/ecm_sdx_pcc.c
@@ -869,7 +869,7 @@ static int ecm_sdx_pcc_rule_read(void *b
  * ecm_sdx_pcc_rule_handler()
  *	Handle rule proc entry
  */
-static int ecm_sdx_pcc_rule_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sdx_pcc_rule_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	if (write) {
 		return ecm_sdx_pcc_rule_write(buffer, lenp, ppos);
@@ -884,7 +884,7 @@ static int ecm_sdx_pcc_rule_handler(stru
  *	Handle unregister proc entry
  *	return 0 for success
  */
-static int ecm_sdx_pcc_unregister_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sdx_pcc_unregister_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_value;
--- a/examples/ecm_sfe_l2.c
+++ b/examples/ecm_sfe_l2.c
@@ -916,7 +916,7 @@ struct ecm_sfe_common_callbacks sfe_cbs
  * ecm_sfe_l2_defunct_by_port_handler()
  * 	Proc handler to defunct the ecm rules by giving port numnber
  */
-static int ecm_sfe_l2_defunct_by_port_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_l2_defunct_by_port_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
@@ -954,7 +954,7 @@ static int ecm_sfe_l2_defunct_by_port_ha
  * ecm_sfe_l2_defunct_by_5tuple_handler()
  * 	Proc handler to defunct the ecm rules by giving 5 tuple
  */
-static int ecm_sfe_l2_defunct_by_5tuple_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_l2_defunct_by_5tuple_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
@@ -1000,7 +1000,7 @@ static int ecm_sfe_l2_defunct_by_5tuple_
  * ecm_sfe_l2_defunct_by_protocol_handler()
  * 	Proc handler to defunct connections based on protocol
  */
-static int ecm_sfe_l2_defunct_by_protocol_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_l2_defunct_by_protocol_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
@@ -1100,7 +1100,7 @@ static int ecm_sfe_l2_policy_rule_read_h
  * Policy rule must include cmd, ip_ver and direction. It can also include src/dest IP and ports, protocol.
  * cmd and ip_ver MUST be the first 2 options in the command.
  */
-static int ecm_sfe_l2_policy_rule_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_l2_policy_rule_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	char *buf;
 	int count;
@@ -1136,7 +1136,7 @@ static int ecm_sfe_l2_policy_rule_handle
  * ecm_sfe_l2_wan_name_handler()
  * 	Writes the WAN interface name to the sysctl node wan_name
  */
-static int ecm_sfe_l2_wan_name_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_l2_wan_name_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int count;
 
--- a/frontends/cmn/ecm_bond_notifier.c
+++ b/frontends/cmn/ecm_bond_notifier.c
@@ -226,7 +226,7 @@ static void ecm_bond_notifier_bond_delet
  * ecm_bond_notifier_stop_handler()
  * 	Proc handler for enable or disable ECM bond notifier
  */
-static int ecm_bond_notifier_stop_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_bond_notifier_stop_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/ecm_front_end_common.c
+++ b/frontends/ecm_front_end_common.c
@@ -933,7 +933,7 @@ uint64_t ecm_front_end_get_slow_packet_c
  * ecm_front_end_ppe_fse_enable_limit_handler()
  *	Sysctl to enable/disable FSE programming through PPE.
  */
-int ecm_front_end_ppe_fse_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_front_end_ppe_fse_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
@@ -968,7 +968,7 @@ int ecm_front_end_ppe_fse_enable_handler
  * ecm_front_end_db_conn_limit_handler()
  *	Database connection limit sysctl node handler.
  */
-int ecm_front_end_db_conn_limit_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+int ecm_front_end_db_conn_limit_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_value;
@@ -1187,7 +1187,7 @@ static int ecm_front_end_denied_ports_ha
  * ecm_front_end_udp_denied_ports_handler()
  *	Proc handler function for UDP denied ports read/write operation.
  */
-static int ecm_front_end_udp_denied_ports_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_front_end_udp_denied_ports_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1207,7 +1207,7 @@ static int ecm_front_end_udp_denied_port
  * ecm_front_end_tcp_denied_ports_handler()
  *	Proc handler function for TCP denied ports read/write operation.
  */
-static int ecm_front_end_tcp_denied_ports_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_front_end_tcp_denied_ports_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/ecm_front_end_ipv4.c
+++ b/frontends/ecm_front_end_ipv4.c
@@ -106,7 +106,7 @@ static void ecm_front_end_ipv4_interface
  * ecm_front_end_ipv4_stop_handler()
  * 	Proc handler to enable or disable ipv4 frontend
  */
-static int ecm_front_end_ipv4_stop_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_front_end_ipv4_stop_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/ecm_front_end_ipv6.c
+++ b/frontends/ecm_front_end_ipv6.c
@@ -107,7 +107,7 @@ static void ecm_front_end_ipv6_interface
  * ecm_front_end_ipv6_stop_handler()
  * 	Proc handler to enable or disable ipv6 frontend
  */
-static int ecm_front_end_ipv6_stop_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_front_end_ipv6_stop_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/ppe/ecm_ppe_ipv4.c
+++ b/frontends/ppe/ecm_ppe_ipv4.c
@@ -757,7 +757,7 @@ static void ecm_ppe_ipv4_sync_queue_exit
  * ecm_ppe_ipv4_accel_limit_mode_handler()
  * 	Proc handler to limit the number of acceleration
  */
-static int ecm_ppe_ipv4_accel_limit_mode_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv4_accel_limit_mode_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -803,7 +803,7 @@ static int ecm_ppe_ipv4_accel_limit_mode
  * ecm_ppe_ipv4_driver_fail_limit_default_handler()
  * 	Proc handler to limit driver interactions failure
  */
-static int ecm_ppe_ipv4_driver_fail_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv4_driver_fail_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -843,7 +843,7 @@ static int ecm_ppe_ipv4_driver_fail_limi
  * ecm_ppe_ipv4_nack_limit_default_handler()
  * 	Proc handler to handle nack limit
  */
-static int ecm_ppe_ipv4_nack_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv4_nack_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -883,7 +883,7 @@ static int ecm_ppe_ipv4_nack_limit_defau
  * ecm_ppe_ipv4_no_action_limit_default_handler()
  * 	Proc handler to set default no action limit
  */
-static int ecm_ppe_ipv4_no_action_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv4_no_action_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -923,7 +923,7 @@ static int ecm_ppe_ipv4_no_action_limit_
  * ecm_ppe_ipv4_vlan_passthrough_enable_handler()
  * 	Proc handler to enable/disable VLAN passthrough
  */
-static int ecm_ppe_ipv4_vlan_passthrough_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv4_vlan_passthrough_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/ppe/ecm_ppe_ipv6.c
+++ b/frontends/ppe/ecm_ppe_ipv6.c
@@ -815,7 +815,7 @@ static void ecm_ppe_ipv6_sync_queue_exit
  * ecm_ppe_ipv4_accel_limit_mode_handler()
  * 	Proc handler to limit the number of connection
  */
-static int ecm_ppe_ipv6_accel_limit_mode_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv6_accel_limit_mode_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -861,7 +861,7 @@ static int ecm_ppe_ipv6_accel_limit_mode
  * ecm_ppe_ipv4_driver_fail_limit_default_handler()
  * 	Proc handler to limit driver interactions failure
  */
-static int ecm_ppe_ipv6_driver_fail_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv6_driver_fail_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -901,7 +901,7 @@ static int ecm_ppe_ipv6_driver_fail_limi
  * ecm_ppe_ipv4_nack_limit_default_handler()
  * 	Proc handler to handle nack limit
  */
-static int ecm_ppe_ipv6_nack_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv6_nack_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -941,7 +941,7 @@ static int ecm_ppe_ipv6_nack_limit_defau
  * ecm_ppe_ipv6_no_action_limit_default_handler()
  * 	Proc handler to set default no action limit
  */
-static int ecm_ppe_ipv6_no_action_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv6_no_action_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -981,7 +981,7 @@ static int ecm_ppe_ipv6_no_action_limit_
  * ecm_ppe_ipv6_vlan_passthrough_enable_handler()
  * 	Proc handler to enable/disable VLAN passthrough
  */
-static int ecm_ppe_ipv6_vlan_passthrough_enable_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_ppe_ipv6_vlan_passthrough_enable_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/sfe/ecm_sfe_common.c
+++ b/frontends/sfe/ecm_sfe_common.c
@@ -338,7 +338,7 @@ void ecm_sfe_common_fast_xmit_set(uint32
  * ecm_sfe_fast_xmit_enable_handler()
  *	Fast transmit sysctl node handler.
  */
-int ecm_sfe_fast_xmit_enable_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+int ecm_sfe_fast_xmit_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
@@ -362,7 +362,7 @@ int ecm_sfe_fast_xmit_enable_handler(str
  * ecm_sfe_fse_enable_handler()
  *	Sysctl to enable/disable FSE programming through ECM SFE frontend.
  */
-int ecm_sfe_fse_enable_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+int ecm_sfe_fse_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_val;
@@ -393,7 +393,7 @@ int ecm_sfe_fse_enable_handler(struct ct
  * ecm_sfe_mht_enable_handler()
  *	Sysctl to enable/disable MHT feature through ECM SFE frontend.
  */
-int ecm_sfe_mht_enable_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+int ecm_sfe_mht_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 	int current_val;
@@ -424,7 +424,7 @@ int ecm_sfe_mht_enable_handler(struct ct
  * ecm_sfe_tun_fast_xmit_enable_handler()
  *	Tunnel fast transmit enable sysctl node handler.
  */
-int ecm_sfe_tun_fast_xmit_enable_handler(struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
+int ecm_sfe_tun_fast_xmit_enable_handler(const struct ctl_table *ctl, int write, void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
 
--- a/frontends/sfe/ecm_sfe_ipv4.c
+++ b/frontends/sfe/ecm_sfe_ipv4.c
@@ -902,7 +902,7 @@ static void ecm_sfe_ipv4_sync_queue_exit
  * ecm_sfe_ipv4_accel_limit_mode_handler()
  * 	Proc handler to limit the number of connection
  */
-static int ecm_sfe_ipv4_accel_limit_mode_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv4_accel_limit_mode_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -948,7 +948,7 @@ static int ecm_sfe_ipv4_accel_limit_mode
  * ecm_sfe_ipv4_driver_fail_limit_default_handler()
  * 	Proc handler to limit driver interactions failure
  */
-static int ecm_sfe_ipv4_driver_fail_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv4_driver_fail_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -988,7 +988,7 @@ static int ecm_sfe_ipv4_driver_fail_limi
  * ecm_sfe_ipv4_nack_limit_default_handler()
  * 	Proc handler to handle nack limit
  */
-static int ecm_sfe_ipv4_nack_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv4_nack_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1028,7 +1028,7 @@ static int ecm_sfe_ipv4_nack_limit_defau
  * ecm_sfe_ipv4_no_action_limit_default_handler()
  * 	Proc handler to set default no action limit
  */
-static int ecm_sfe_ipv4_no_action_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv4_no_action_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1069,7 +1069,7 @@ static int ecm_sfe_ipv4_no_action_limit_
  * ecm_sfe_ipv4_reject_acceleration_for_ipsec_handler()
  * 	Proc handler to enable/diable IPsec acceleration
  */
-static int ecm_sfe_ipv4_reject_acceleration_for_ipsec_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv4_reject_acceleration_for_ipsec_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/sfe/ecm_sfe_ipv6.c
+++ b/frontends/sfe/ecm_sfe_ipv6.c
@@ -898,7 +898,7 @@ static void ecm_sfe_ipv6_sync_queue_exit
  * ecm_sfe_ipv6_accel_limit_mode_handler()
  * 	Proc handler to limit the number of connection
  */
-static int ecm_sfe_ipv6_accel_limit_mode_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv6_accel_limit_mode_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -944,7 +944,7 @@ static int ecm_sfe_ipv6_accel_limit_mode
  * ecm_sfe_ipv6_driver_fail_limit_default_handler()
  * 	Proc handler to limit driver interactions failure
  */
-static int ecm_sfe_ipv6_driver_fail_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv6_driver_fail_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -984,7 +984,7 @@ static int ecm_sfe_ipv6_driver_fail_limi
  * ecm_sfe_ipv6_nack_limit_default_handler()
  * 	Proc handler to handle nack limit
  */
-static int ecm_sfe_ipv6_nack_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv6_nack_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1024,7 +1024,7 @@ static int ecm_sfe_ipv6_nack_limit_defau
  * ecm_sfe_ipv6_no_action_limit_default_handler()
  * 	Proc handler to set default no action limit
  */
-static int ecm_sfe_ipv6_no_action_limit_default_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv6_no_action_limit_default_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -1065,7 +1065,7 @@ static int ecm_sfe_ipv6_no_action_limit_
  * ecm_sfe_ipv6_reject_acceleration_for_ipsec_handler()
  * 	Proc handler to enable/disable IPsec acceleration
  */
-static int ecm_sfe_ipv6_reject_acceleration_for_ipsec_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_sfe_ipv6_reject_acceleration_for_ipsec_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/sfe/ecm_sfe_multicast_ipv4.c
+++ b/frontends/sfe/ecm_sfe_multicast_ipv4.c
@@ -3180,7 +3180,7 @@ find_next_tuple:
  * ecm_front_end_ipv4_mc_stop_handler()
  * 	Proc handler to enable/disable multicast traffic via SFE
  */
-static int ecm_front_end_ipv4_mc_stop_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_front_end_ipv4_mc_stop_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
--- a/frontends/sfe/ecm_sfe_multicast_ipv6.c
+++ b/frontends/sfe/ecm_sfe_multicast_ipv6.c
@@ -3173,7 +3173,7 @@ static void ecm_sfe_multicast_ipv6_mfc_u
  * ecm_front_end_ipv6_mc_stop_handler()
  * 	Proc handler to enable/disable IPv6 Mulitcast via SFE
  */
-static int ecm_front_end_ipv6_mc_stop_handler(struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_front_end_ipv6_mc_stop_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
