--- a/ecm_db/ecm_db_node.c
+++ b/ecm_db/ecm_db_node.c
@@ -1214,6 +1214,7 @@ void ecm_db_traverse_node_connection_lis
 							    ecm_db_connection_defunct_type_t type,
 							    uint8_t wifi_qm_type, uint8_t wifi_qm_id)
 {
+#ifdef ECM_CLASSIFIER_WIFI_ENABLE
 	struct ecm_db_connection_instance *ci = NULL;
 	struct ecm_db_connection_defunct_info info;
 	struct ecm_classifier_instance *aci;
@@ -1257,6 +1258,7 @@ next_conn:
 		ci = cin;
 	}
 	DEBUG_INFO("%px: Defuncting from node connection list by qm complete\n", node);
+#endif
 }
 
 /*
@@ -1265,6 +1267,7 @@ next_conn:
  */
 void ecm_db_node_defunct_qm_connections(uint8_t *mac, uint8_t wifi_qm_type, uint8_t wifi_qm_id)
 {
+#ifdef ECM_CLASSIFIER_WIFI_ENABLE
 	struct ecm_db_node_instance *ni = NULL;
 
 	DEBUG_INFO("Defunct connections assigned to wifi qm based on qm_type and qm_id\n");
@@ -1318,6 +1321,7 @@ void ecm_db_node_defunct_qm_connections(
 #ifdef ECM_IPV6_ENABLE
 	ecm_front_end_ipv6_stop_temp(0);
 #endif
+#endif
 }
 
 /*
