--- a/ecm_classifier_default.c
+++ b/ecm_classifier_default.c
@@ -819,7 +819,6 @@ static struct ctl_table ecm_classifier_d
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_accel_delay_pkts_handler,
 	},
-	{ }
 };
 
 /*
--- a/ecm_compat.h
+++ b/ecm_compat.h
@@ -133,6 +133,7 @@ static inline void ecm_compat_ndisc_send
 #define ndisc_send_ns(dev, solicit, daddr, saddr, nonce) ecm_compat_ndisc_send_ns(dev, solicit, daddr, saddr, nonce)
 
 /* ip_route_output Compat */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 10, 0)
 static inline struct rtable *ecm_compat_ip_route_output(struct net *net, __be32 daddr, __be32 saddr, u8 tos, int oif)
 {
        struct flowi4 fl4 = {
@@ -144,6 +145,7 @@ static inline struct rtable *ecm_compat_
        return ip_route_output_flow(net, &fl4, NULL);
 }
 #define ip_route_output(net, daddr, saddr, tos, oif) ecm_compat_ip_route_output(net, daddr, saddr, tos, oif)
+#endif
 
 /* Neigh Mac Update Compat - QCA Specific */
 /* Struct definition required for dead code compatibility in ecm_interface.c */
--- a/ecm_db/ecm_db.c
+++ b/ecm_db/ecm_db.c
@@ -21,6 +21,8 @@
 #include <net/route.h>
 #include <net/ip.h>
 #include <net/tcp.h>
+#include <net/ip_fib.h>
+#include <net/fib_notifier.h>
 #include <asm/unaligned.h>
 #include <asm/uaccess.h>	/* for put_user */
 #include <net/ipv6.h>
@@ -669,7 +671,6 @@ static struct ctl_table ecm_db_ctl_table
 	},
 #endif
 
-	{ }
 };
 
 /*
--- a/ecm_db/ecm_db_node.c
+++ b/ecm_db/ecm_db_node.c
@@ -1214,7 +1214,6 @@ void ecm_db_traverse_node_connection_lis
 							    ecm_db_connection_defunct_type_t type,
 							    uint8_t wifi_qm_type, uint8_t wifi_qm_id)
 {
-#ifdef ECM_CLASSIFIER_WIFI_ENABLE
 	struct ecm_db_connection_instance *ci = NULL;
 	struct ecm_db_connection_defunct_info info;
 	struct ecm_classifier_instance *aci;
@@ -1258,7 +1257,6 @@ next_conn:
 		ci = cin;
 	}
 	DEBUG_INFO("%px: Defuncting from node connection list by qm complete\n", node);
-#endif
 }
 
 /*
@@ -1267,7 +1265,6 @@ next_conn:
  */
 void ecm_db_node_defunct_qm_connections(uint8_t *mac, uint8_t wifi_qm_type, uint8_t wifi_qm_id)
 {
-#ifdef ECM_CLASSIFIER_WIFI_ENABLE
 	struct ecm_db_node_instance *ni = NULL;
 
 	DEBUG_INFO("Defunct connections assigned to wifi qm based on qm_type and qm_id\n");
@@ -1321,7 +1318,6 @@ void ecm_db_node_defunct_qm_connections(
 #ifdef ECM_IPV6_ENABLE
 	ecm_front_end_ipv6_stop_temp(0);
 #endif
-#endif
 }
 
 /*
--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -1637,7 +1637,11 @@ __be32 ecm_interface_vxlan_gpe_get_vni_r
 		__be32 daddr;
 
 		daddr = netif_is_vxlan(indev) ? ip_hdr(skb)->saddr : ip_hdr(skb)->daddr;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 10, 0)
 		rt = ip_route_output(&init_net, daddr, 0, 0, 0);
+#else
+		rt = ip_route_output(&init_net, daddr, 0, 0, 0, 0);
+#endif
 		if (IS_ERR_OR_NULL(rt)) {
 			DEBUG_WARN("%px: VXLAN-GPE failed to get IPv4 route to: %pI4\n", dev, &daddr);
 			goto rt_error;
@@ -1960,7 +1964,11 @@ struct neighbour *ecm_interface_ipv4_nei
 	__be32 ipv4_addr;
 
 	ECM_IP_ADDR_TO_NIN4_ADDR(ipv4_addr, addr);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 10, 0)
 	rt = ip_route_output(&init_net, ipv4_addr, 0, 0, 0);
+#else
+	rt = ip_route_output(&init_net, ipv4_addr, 0, 0, 0, 0);
+#endif
 	if (IS_ERR_OR_NULL(rt)) {
 		return NULL;
 	}
@@ -9599,7 +9607,6 @@ static struct ctl_table ecm_interface_ta
 		.mode			= 0644,
 		.proc_handler		= &ecm_interface_defunct_by_iface_handler,
 	},
-	{ }
 };
 
 #ifdef ECM_INTERFACE_IPSEC_GLUE_LAYER_SUPPORT_ENABLE
--- a/frontends/sfe/ecm_sfe_common.c
+++ b/frontends/sfe/ecm_sfe_common.c
@@ -528,7 +528,6 @@ static struct ctl_table ecm_sfe_sysctl_t
 		.mode		= 0644,
 		.proc_handler	= &ecm_sfe_tun_fast_xmit_enable_handler,
 	},
-	{}
 };
 
 /*
@@ -953,6 +952,7 @@ bool ecm_sfe_common_get_mht_port_id(stru
 	 * SFE acceleration to avoid HOLB issue. While DSA interfaces mark the skb->mark at the DSA TX datapath,
 	 * here it doesn't need to query the switch FDB to get the skb->mark.
 	 */
+#if defined(ECM_MHT_ENABLE) && defined(ECM_FRONT_END_PPE_ENABLE)
 	if (ppe_drv_is_mht_dev(dev) && (ecm_db_iface_type_get(from_sfe_iface) != ECM_DB_IFACE_TYPE_DSA)) {
 		ecm_db_connection_node_address_get(feci->ci, ECM_DB_OBJ_DIR_FROM, mht_mac);
 		port_info = ppe_drv_mht_port_from_fdb(mht_mac, 0);
@@ -966,6 +966,7 @@ bool ecm_sfe_common_get_mht_port_id(stru
 			*valid_flags |= SFE_RULE_CREATE_MARK_VALID;
 			return true;
 		}
+#endif
 
 		spin_lock_bh(&feci->lock);
 		/*
@@ -1000,6 +1001,7 @@ bool ecm_sfe_common_get_mht_port_id(stru
 	 * SFE acceleration to avoid HOLB issue. While DSA interfaces mark the skb->mark at the DSA TX datapath,
 	 * here it doesn't need to query the switch FDB to get the skb->mark.
 	 */
+#if defined(ECM_MHT_ENABLE) && defined(ECM_FRONT_END_PPE_ENABLE)
 	if (ppe_drv_is_mht_dev(dev) && (ecm_db_iface_type_get(to_sfe_iface) != ECM_DB_IFACE_TYPE_DSA)) {
 		ecm_db_connection_node_address_get(feci->ci, ECM_DB_OBJ_DIR_TO, mht_mac);
 		port_info = ppe_drv_mht_port_from_fdb(mht_mac, 0);
@@ -1013,6 +1015,7 @@ bool ecm_sfe_common_get_mht_port_id(stru
 			*valid_flags |= SFE_RULE_CREATE_MARK_VALID;
 			return true;
 		}
+#endif
 
 		spin_lock_bh(&feci->lock);
 		/*
@@ -1056,7 +1059,9 @@ int ecm_sfe_common_get_vp_from_iface_id(
 
 #ifdef ECM_INTERFACE_GRE_TAP_ENABLE
 	if (dev->priv_flags_ext & (IFF_EXT_GRE_V4_TAP | IFF_EXT_GRE_V6_TAP)) {
+#ifdef ECM_FRONT_END_PPE_ENABLE
 		vp = ppe_tun_hybrid_ol_ctx_get(dev);
+#endif
 	}
 #endif
 	dev_put(dev);
