--- a/ecm_classifier.c
+++ b/ecm_classifier.c
@@ -75,7 +75,7 @@
  *  1 - Acceleration will not start until both direction traffic is seen.
  *  N - Acceleration will not start until N packets are seen in the slow path.
  */
-int ecm_classifier_accel_delay_pkts = 0;
+int ecm_classifier_accel_delay_pkts_v2 = 0;
 
 /*
  * ecm_classifier_assign_classifier()
--- a/ecm_classifier.h
+++ b/ecm_classifier.h
@@ -19,7 +19,7 @@
 
 #define ECM_CLASSIFIER_UDP_IPSEC_PORT 5200	/* Default port for UDP IPsec */
 
-extern int ecm_classifier_accel_delay_pkts;	/* Default slow path packets allowed before the acceleration */
+extern int ecm_classifier_accel_delay_pkts_v2;	/* Default slow path packets allowed before the acceleration */
 
 struct ecm_classifier_instance;
 
--- a/ecm_classifier_default.c
+++ b/ecm_classifier_default.c
@@ -231,7 +231,7 @@ static bool ecm_classifier_default_ready
 	/*
 	 * Delay forever until seeing the reply packet.
 	 */
-	if (ecm_classifier_accel_delay_pkts == 1) {
+	if (ecm_classifier_accel_delay_pkts_v2 == 1) {
 		DEBUG_INFO("%px: Checking both direction traffic\n", cdii);
 
 		/*
@@ -251,7 +251,7 @@ static bool ecm_classifier_default_ready
 
 	/*
 	 * Delay the acceleration until we see <N> number of packets.
-	 * ecm_classifier_accel_delay_pkts = <N>
+	 * ecm_classifier_accel_delay_pkts_v2 = <N>
 	 */
 	ci = ecm_db_connection_serial_find_and_ref(cdii->ci_serial);
 	if (!ci) {
@@ -270,9 +270,9 @@ static bool ecm_classifier_default_ready
 	/*
 	 * Check if we have seen slow path packets as the predefined count.
 	 */
-	if (slow_pkts < ecm_classifier_accel_delay_pkts) {
+	if (slow_pkts < ecm_classifier_accel_delay_pkts_v2) {
 		DEBUG_TRACE("%px: delay the acceleration: slow packets: %llu default delay packet count: %d\n",
-			    cdii, slow_pkts, ecm_classifier_accel_delay_pkts);
+			    cdii, slow_pkts, ecm_classifier_accel_delay_pkts_v2);
 
 		/*
 		 * We haven't reached the slow path packet limit.
@@ -440,7 +440,7 @@ check_delay:
 	/*
 	 * Should we delay the acceleration?
 	 */
-	if (ecm_classifier_accel_delay_pkts) {
+	if (ecm_classifier_accel_delay_pkts_v2) {
 		if (!ecm_classifier_default_ready_for_accel(cdii, sender)) {
 			DEBUG_INFO("%px: connection is not ready for accel\n", cdii);
 			spin_lock_bh(&ecm_classifier_default_lock);
@@ -621,7 +621,7 @@ static int ecm_classifier_default_state_
 	if ((result = ecm_state_write(sfi, "timer_group", "%d", timer_group))) {
 		return result;
 	}
-	if ((result = ecm_state_write(sfi, "accel_delay_pkt_default", "%d", ecm_classifier_accel_delay_pkts))) {
+	if ((result = ecm_state_write(sfi, "accel_delay_pkt_default", "%d", ecm_classifier_accel_delay_pkts_v2))) {
 		return result;
 	}
 
@@ -750,11 +750,11 @@ static int ecm_classifier_default_accel_
 }
 
 /*
- * ecm_classifier_accel_delay_pkts_handler()
+ * ecm_classifier_accel_delay_pkts_v2_handler()
  * 	Proc handler to handle slow path packets allowed
  * 	before acceleration
  */
-static int ecm_classifier_accel_delay_pkts_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
+static int ecm_classifier_accel_delay_pkts_v2_handler(const struct ctl_table *ctl, int write, void *buffer, size_t *lenp, loff_t *ppos)
 {
 	/*
 	 * Usage:
@@ -779,7 +779,7 @@ static int ecm_classifier_accel_delay_pk
 	/*
 	 * Write the value with user input
 	 */
-	current_val = ecm_classifier_accel_delay_pkts;
+	current_val = ecm_classifier_accel_delay_pkts_v2;
 	ret = proc_dointvec(ctl, write, buffer, lenp, ppos);
 	if (ret || (!write)) {
 		/*
@@ -788,8 +788,8 @@ static int ecm_classifier_accel_delay_pk
 		return ret;
 	}
 
-	if (ecm_classifier_accel_delay_pkts < 0) {
-		ecm_classifier_accel_delay_pkts = current_val;
+	if (ecm_classifier_accel_delay_pkts_v2 < 0) {
+		ecm_classifier_accel_delay_pkts_v2 = current_val;
 		DEBUG_ERROR("Invalid input, Valid inout should be positive\n");
 		return -EINVAL;
 	}
@@ -813,11 +813,11 @@ static struct ctl_table ecm_classifier_d
 		.proc_handler	= &ecm_classifier_default_accel_mode_handler,
 	},
 	{
-		.procname	= "accel_delay_pkts",
-		.data		= &ecm_classifier_accel_delay_pkts,
+		.procname	= "accel_delay_pkts_v2",
+		.data		= &ecm_classifier_accel_delay_pkts_v2,
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
-		.proc_handler	= &ecm_classifier_accel_delay_pkts_handler,
+		.proc_handler	= &ecm_classifier_accel_delay_pkts_v2_handler,
 	},
 };
 
@@ -970,8 +970,8 @@ int ecm_classifier_default_init(struct d
 					(u32 *)&ecm_classifier_default_enabled);
 	ecm_debugfs_create_u32("accel_mode", S_IRUGO | S_IWUSR, ecm_classifier_default_dentry,
 					(u32 *)&ecm_classifier_default_accel_mode);
-	ecm_debugfs_create_u32("accel_delay_pkts", S_IRUGO | S_IWUSR, ecm_classifier_default_dentry,
-					(u32 *)&ecm_classifier_accel_delay_pkts);
+	ecm_debugfs_create_u32("accel_delay_pkts_v2", S_IRUGO | S_IWUSR, ecm_classifier_default_dentry,
+					(u32 *)&ecm_classifier_accel_delay_pkts_v2);
 
 	return 0;
 }
--- a/ecm_classifier_dscp.c
+++ b/ecm_classifier_dscp.c
@@ -461,7 +461,7 @@ static void ecm_classifier_dscp_process(
 		 * Option 2: Wait for seeing N number of packets. If we still don't see reply,
 		 * set the uni-directional values.
 		 */
-		if (ecm_classifier_accel_delay_pkts) {
+		if (ecm_classifier_accel_delay_pkts_v2) {
 			/*
 			 * Stop the processing if both side packets are already seen.
 			 * Above the process response is already set to allow the
@@ -485,11 +485,11 @@ static void ecm_classifier_dscp_process(
 			/*
 			 * Deny the acceleration if any of the below options holds true.
 			 * For option 1, we wait forever
-			 * For option 2, we wait until seeing ecm_classifier_accel_delay_pkts.
+			 * For option 2, we wait until seeing ecm_classifier_accel_delay_pkts_v2.
 			 */
-			if ((ecm_classifier_accel_delay_pkts == 1) || (slow_pkts < ecm_classifier_accel_delay_pkts)) {
+			if ((ecm_classifier_accel_delay_pkts_v2 == 1) || (slow_pkts < ecm_classifier_accel_delay_pkts_v2)) {
 				DEBUG_TRACE("%px: accel_delay_pkts: %d slow_pkts: %llu accel is not allowed yet\n",
-						cdscpi, ecm_classifier_accel_delay_pkts, slow_pkts);
+						cdscpi, ecm_classifier_accel_delay_pkts_v2, slow_pkts);
 				cdscpi->process_response.accel_mode = ECM_CLASSIFIER_ACCELERATION_MODE_NO;
 				goto dscp_classifier_out;
 			}
--- a/ecm_classifier_emesh.c
+++ b/ecm_classifier_emesh.c
@@ -2245,7 +2245,7 @@ sawf_classifier_out:
 		 * Option 2: Wait for seeing N number of packets. If we still don't see reply,
 		 * set the uni-directional values.
 		 */
-		if (ecm_classifier_accel_delay_pkts) {
+		if (ecm_classifier_accel_delay_pkts_v2) {
 			/*
 			 * Stop the processing if both side packets are already seen.
 			 * Above the process response is already set to allow the
@@ -2269,11 +2269,11 @@ sawf_classifier_out:
 			/*
 			 * Deny the acceleration if any of the below options holds true.
 			 * For option 1, we wait forever
-			 * For option 2, we wait until seeing ecm_classifier_accel_delay_pkts.
+			 * For option 2, we wait until seeing ecm_classifier_accel_delay_pkts_v2.
 			 */
-			if ((ecm_classifier_accel_delay_pkts == 1) || (slow_pkts < ecm_classifier_accel_delay_pkts)) {
+			if ((ecm_classifier_accel_delay_pkts_v2 == 1) || (slow_pkts < ecm_classifier_accel_delay_pkts_v2)) {
 				DEBUG_TRACE("%px: accel_delay_pkts: %d slow_pkts: %llu accel is not allowed yet\n",
-						cemi, ecm_classifier_accel_delay_pkts, slow_pkts);
+						cemi, ecm_classifier_accel_delay_pkts_v2, slow_pkts);
 				spin_lock_bh(&ecm_classifier_emesh_sawf_lock);
 				cemi->process_response.accel_mode = ECM_CLASSIFIER_ACCELERATION_MODE_NO;
 				goto sawf_emesh_classifier_out;
--- a/ecm_classifier_wifi.c
+++ b/ecm_classifier_wifi.c
@@ -382,7 +382,7 @@ static void ecm_classifier_wifi_process(
 			goto process_wifi_classifier_out;
 		}
 
-		if (ecm_classifier_accel_delay_pkts) {
+		if (ecm_classifier_accel_delay_pkts_v2) {
 
 			/*
 			 * Store the PCP value in the classifier instance and
@@ -390,9 +390,9 @@ static void ecm_classifier_wifi_process(
 			 */
 			ecm_classifier_wifi_fill_pcp(cwifii, sender, skb);
 
-			if ((ecm_classifier_accel_delay_pkts == 1) || (slow_pkts < ecm_classifier_accel_delay_pkts)) {
+			if ((ecm_classifier_accel_delay_pkts_v2 == 1) || (slow_pkts < ecm_classifier_accel_delay_pkts_v2)) {
 				DEBUG_TRACE("%px: accel_delay_pkts: %d slow_pkts: %llu accel is not allowed yet\n",
-						cwifii, ecm_classifier_accel_delay_pkts, slow_pkts);
+						cwifii, ecm_classifier_accel_delay_pkts_v2, slow_pkts);
 				cwifii->process_response.accel_mode = ECM_CLASSIFIER_ACCELERATION_MODE_NO;
 				goto process_wifi_classifier_out;
 			}
