--- a/ecm_classifier_mark.c
+++ b/ecm_classifier_mark.c
@@ -675,8 +675,7 @@ static struct ctl_table ecm_classifier_m
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_mark_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_classifier_dscp.c
+++ b/ecm_classifier_dscp.c
@@ -944,8 +944,7 @@ static struct ctl_table ecm_classifier_d
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_dscp_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_classifier_emesh.c
+++ b/ecm_classifier_emesh.c
@@ -4376,8 +4376,7 @@ static struct ctl_table ecm_classifier_e
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_sawf_emesh_udp_ipsec_port_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/frontends/ecm_front_end_ipv4.c
+++ b/frontends/ecm_front_end_ipv4.c
@@ -151,8 +151,7 @@ static struct ctl_table ecm_front_end_ip
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_front_end_ipv4_stop_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/frontends/sfe/ecm_sfe_ipv4.c
+++ b/frontends/sfe/ecm_sfe_ipv4.c
@@ -1144,7 +1144,6 @@ static struct ctl_table ecm_sfe_ipv4_ctl
 		.proc_handler	= &ecm_sfe_ipv4_reject_acceleration_for_ipsec_handler,
 	},
 #endif
-	{ }
 };
 
 /*
--- a/frontends/sfe/ecm_sfe_ipv6.c
+++ b/frontends/sfe/ecm_sfe_ipv6.c
@@ -1143,7 +1143,6 @@ static struct ctl_table ecm_sfe_ipv6_ctl
 		.proc_handler	= &ecm_sfe_ipv6_reject_acceleration_for_ipsec_handler,
 	},
 #endif
-	{ }
 };
 
 /*
--- a/frontends/sfe/ecm_sfe_multicast_ipv4.c
+++ b/frontends/sfe/ecm_sfe_multicast_ipv4.c
@@ -3226,8 +3226,7 @@ static struct ctl_table ecm_sfe_multicas
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_front_end_ipv4_mc_stop_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/frontends/sfe/ecm_sfe_multicast_ipv6.c
+++ b/frontends/sfe/ecm_sfe_multicast_ipv6.c
@@ -3219,8 +3219,7 @@ static struct ctl_table ecm_sfe_multicas
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_front_end_ipv6_mc_stop_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/frontends/ecm_front_end_ipv6.c
+++ b/frontends/ecm_front_end_ipv6.c
@@ -152,8 +152,7 @@ static struct ctl_table ecm_front_end_ip
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_front_end_ipv6_stop_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/frontends/cmn/ecm_bond_notifier.c
+++ b/frontends/cmn/ecm_bond_notifier.c
@@ -269,8 +269,7 @@ static struct ctl_table ecm_bond_notifie
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_bond_notifier_stop_handler,
-	},
-	{ }
+	}
 };
 
 void ecm_bond_notifier_stop(int num)
--- a/frontends/ppe/ecm_ppe_ipv6.c
+++ b/frontends/ppe/ecm_ppe_ipv6.c
@@ -1055,8 +1055,7 @@ static struct ctl_table ecm_ppe_ipv6_ctl
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_ppe_ipv6_vlan_passthrough_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_classifier_mscs.c
+++ b/ecm_classifier_mscs.c
@@ -1420,7 +1420,6 @@ static struct ctl_table ecm_classifier_m
 		.proc_handler	= &ecm_classifier_mscs_scs_udp_ipsec_port_handler,
 	},
 #endif
-	{ }
 };
 
 /*
--- a/ecm_classifier_pcc.c
+++ b/ecm_classifier_pcc.c
@@ -1366,8 +1366,7 @@ static struct ctl_table ecm_classifier_p
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_pcc_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_classifier_ovs.c
+++ b/ecm_classifier_ovs.c
@@ -2230,8 +2230,7 @@ static struct ctl_table ecm_classifier_o
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_ovs_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_state.c
+++ b/ecm_state.c
@@ -971,8 +971,7 @@ static struct ctl_table ecm_state_ctl_ta
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_state_file_output_mask_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_conntrack_notifier.c
+++ b/ecm_conntrack_notifier.c
@@ -449,8 +449,7 @@ static struct ctl_table ecm_conntrack_no
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_conntrack_notifier_stop_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/frontends/ppe/ecm_ppe_ipv4.c
+++ b/frontends/ppe/ecm_ppe_ipv4.c
@@ -997,8 +997,7 @@ static struct ctl_table ecm_ppe_ipv4_ctl
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_ppe_ipv4_vlan_passthrough_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_db/ecm_db_connection.c
+++ b/ecm_db/ecm_db_connection.c
@@ -5270,8 +5270,7 @@ static struct ctl_table ecm_db_connectio
 		.maxlen		= sizeof(int),
 		.mode		= 0666,
 		.proc_handler	= &ecm_db_connection_defunct_by_5tuple_mask_handler,
-	},
-	{ }
+	}
 };
 
 /*
--- a/ecm_classifier_wifi.c
+++ b/ecm_classifier_wifi.c
@@ -721,8 +721,7 @@ static struct ctl_table ecm_classifier_w
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &ecm_classifier_wifi_enable_handler,
-	},
-	{ }
+	}
 };
 
 /*
