--- a/ecm_classifier_default.c
+++ b/ecm_classifier_default.c
@@ -797,7 +797,7 @@ static int ecm_classifier_accel_delay_pk
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_default_ctl_table[] = {
+static const struct ctl_table ecm_classifier_default_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_default_enabled,
--- a/ecm_classifier_dscp.c
+++ b/ecm_classifier_dscp.c
@@ -937,7 +937,7 @@ static int ecm_classifier_dscp_enable_ha
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_dscp_ctl_table[] = {
+static const struct ctl_table ecm_classifier_dscp_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_dscp_enabled,
--- a/ecm_classifier_emesh.c
+++ b/ecm_classifier_emesh.c
@@ -4334,7 +4334,7 @@ static int ecm_classifier_sawf_emesh_udp
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_emesh_ctl_table[] = {
+static const struct ctl_table ecm_classifier_emesh_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_emesh_enabled,
--- a/ecm_classifier_mark.c
+++ b/ecm_classifier_mark.c
@@ -668,7 +668,7 @@ static int ecm_classifier_mark_enable_ha
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_mark_ctl_table[] = {
+static const struct ctl_table ecm_classifier_mark_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_mark_enabled,
--- a/ecm_classifier_mscs.c
+++ b/ecm_classifier_mscs.c
@@ -1389,7 +1389,7 @@ static int ecm_classifier_mscs_scs_udp_i
 }
 #endif
 
-static struct ctl_table ecm_classifier_mscs_ctl_table[] = {
+static const struct ctl_table ecm_classifier_mscs_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_mscs_enabled,
--- a/ecm_classifier_ovs.c
+++ b/ecm_classifier_ovs.c
@@ -2223,7 +2223,7 @@ static int ecm_classifier_ovs_enable_han
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_ovs_ctl_table[] = {
+static const struct ctl_table ecm_classifier_ovs_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_ovs_enabled,
--- a/ecm_classifier_pcc.c
+++ b/ecm_classifier_pcc.c
@@ -1359,7 +1359,7 @@ static int ecm_classifier_pcc_enable_han
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_pcc_ctl_table[] = {
+static const struct ctl_table ecm_classifier_pcc_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_pcc_enabled,
--- a/ecm_classifier_wifi.c
+++ b/ecm_classifier_wifi.c
@@ -714,7 +714,7 @@ static int ecm_classifier_wifi_enable_ha
 	return ret;
 }
 
-static struct ctl_table ecm_classifier_wifi_ctl_table[] = {
+static const struct ctl_table ecm_classifier_wifi_ctl_table[] = {
 	{
 		.procname	= "enabled",
 		.data		= &ecm_classifier_wifi_enabled,
--- a/ecm_conntrack_notifier.c
+++ b/ecm_conntrack_notifier.c
@@ -442,7 +442,7 @@ static struct nf_ct_event_notifier ecm_c
 #endif
 #endif
 
-static struct ctl_table ecm_conntrack_notifier_ctl_table[] = {
+static const struct ctl_table ecm_conntrack_notifier_ctl_table[] = {
 	{
 		.procname	= "stop",
 		.data		= &ecm_conntrack_notifier_stopped,
--- a/ecm_db/ecm_db.c
+++ b/ecm_db/ecm_db.c
@@ -675,7 +675,7 @@ static int ecm_db_ipv6_route_handler(con
 }
 #endif
 
-static struct ctl_table ecm_db_ctl_table[] = {
+static const struct ctl_table ecm_db_ctl_table[] = {
 	{
 		.procname	= "defunct_all",
 		.data		= NULL,
--- a/ecm_db/ecm_db_connection.c
+++ b/ecm_db/ecm_db_connection.c
@@ -5256,7 +5256,7 @@ static int ecm_db_connection_defunct_by_
 	return count;
 }
 
-static struct ctl_table ecm_db_connection_ctl_table[] = {
+static const struct ctl_table ecm_db_connection_ctl_table[] = {
 	{
 		.procname	= "defunct_by_5tuple",
 		.data		= &ecm_db_connection_defunct_5tuple,
--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -9545,7 +9545,7 @@ static int ecm_interface_defunct_by_ifac
 	return ecm_interface_defunct_by_iface(write, buffer, lenp, ppos);
 }
 
-static struct ctl_table ecm_interface_table[] = {
+static const struct ctl_table ecm_interface_table[] = {
 	{
 		.procname		= "src_interface_check",
 		.data			= &ecm_interface_src_check,
--- a/ecm_sdx/ecm_sdx.c
+++ b/ecm_sdx/ecm_sdx.c
@@ -189,7 +189,7 @@ static int ecm_sdx_lan_prefix_handler(co
 	return ecm_sdx_stats_lan_prefix_handler_read(buffer, lenp, ppos);
 }
 
-static struct ctl_table ecm_sdx_sysctl_tbl[] = {
+static const struct ctl_table ecm_sdx_sysctl_tbl[] = {
 	{
 		.procname	= "stats",
 		.data		= NULL,
--- a/ecm_state.c
+++ b/ecm_state.c
@@ -957,7 +957,7 @@ static int ecm_state_file_output_mask_ha
 	return ret;
 }
 
-static struct ctl_table ecm_state_ctl_table[] = {
+static const struct ctl_table ecm_state_ctl_table[] = {
 	{
 		.procname	= "state_dev_major",
 		.data		= &ecm_state_dev_major_id,
--- a/ecm_tracker_udp.c
+++ b/ecm_tracker_udp.c
@@ -1204,7 +1204,7 @@ int ecm_tracker_udp_clf_enabled_handler(
 	return ret;
 }
 
-static struct ctl_table ecm_tracker_udp_sysctl_tbl[] = {
+static const struct ctl_table ecm_tracker_udp_ctl_table[] = {
 	{
 		.procname	= "udp_classification_enabled",
 		.data		= &ecm_tracker_udp_clf_enabled,
@@ -1308,7 +1308,7 @@ EXPORT_SYMBOL(ecm_tracker_udp_alloc);
  */
 void ecm_tracker_udp_sysctl_register(void)
 {
-	ecm_tracker_udp_ctl_tbl_hdr = register_sysctl("net/ecm", ecm_tracker_udp_sysctl_tbl);
+	ecm_tracker_udp_ctl_tbl_hdr = register_sysctl("net/ecm", ecm_tracker_udp_ctl_table);
 }
 
 /*
--- a/examples/ecm_ae_select.c
+++ b/examples/ecm_ae_select.c
@@ -693,7 +693,7 @@ static int ecm_ae_select_test_rule_handl
 /*
  * Sysctl table entried
  */
-static struct ctl_table ecm_ae_select_test_ctl_table[] = {
+static const struct ctl_table ecm_ae_select_test_ctl_table[] = {
 	{
 		.procname	= "rule",
 		.data		= NULL,
--- a/examples/ecm_pcc_test.c
+++ b/examples/ecm_pcc_test.c
@@ -1537,7 +1537,7 @@ static int ecm_pcc_test_pcc_mirror_en_ha
 	return ret;
 }
 
-static struct ctl_table ecm_pcc_test_ctl_table[] = {
+static const struct ctl_table ecm_pcc_test_ctl_table[] = {
 	{
 		.procname	= "unregister",
 		.data		= NULL,
--- a/examples/ecm_sdx_pcc.c
+++ b/examples/ecm_sdx_pcc.c
@@ -921,7 +921,7 @@ static int ecm_sdx_pcc_unregister_handle
 	return 0;
 }
 
-static struct ctl_table ecm_sdx_pcc_sysctl_tbl[] = {
+static const struct ctl_table ecm_sdx_pcc_sysctl_tbl[] = {
 	{
 		.procname		= "ecm_sdx_pcc_rule",
 		.data			= &ecm_sdx_pcc_rules,
--- a/examples/ecm_sfe_l2.c
+++ b/examples/ecm_sfe_l2.c
@@ -1175,7 +1175,7 @@ static int ecm_sfe_l2_wan_name_handler(c
 	return 0;
 }
 
-static struct ctl_table ecm_sfe_l2_ctl_table[] = {
+static const struct ctl_table ecm_sfe_l2_ctl_table[] = {
 	{
 		.procname	= "wan_name",
 		.data		= NULL,
--- a/frontends/cmn/ecm_bond_notifier.c
+++ b/frontends/cmn/ecm_bond_notifier.c
@@ -262,7 +262,7 @@ static int ecm_bond_notifier_stop_handle
 	return ret;
 }
 
-static struct ctl_table ecm_bond_notifier_ctl_table[] = {
+static const struct ctl_table ecm_bond_notifier_ctl_table[] = {
 	{
 		.procname	= "stop",
 		.data		= &ecm_bond_notifier_stopped,
--- a/frontends/ecm_front_end_common.c
+++ b/frontends/ecm_front_end_common.c
@@ -1223,7 +1223,7 @@ static int ecm_front_end_tcp_denied_port
 	return ecm_front_end_denied_ports_handler(write, buffer, lenp, ppos, ecm_front_end_tcp_denied_ports, false);
 }
 
-static struct ctl_table ecm_front_end_sysctl_tbl[] = {
+static const struct ctl_table ecm_front_end_sysctl_table[] = {
 	{
 		.procname	= "front_end_conn_limit",
 		.data		= &ecm_front_end_conn_limit,
@@ -1266,7 +1266,7 @@ void ecm_front_end_common_sysctl_registe
 	/*
 	 * Register sysctl table.
 	 */
-	ecm_front_end_ctl_tbl_hdr = register_sysctl("net/ecm", ecm_front_end_sysctl_tbl);
+	ecm_front_end_ctl_tbl_hdr = register_sysctl("net/ecm", ecm_front_end_sysctl_table);
 #ifdef ECM_FRONT_END_SFE_ENABLE
 	if (ecm_front_end_ctl_tbl_hdr) {
 		ecm_sfe_sysctl_tbl_init();
--- a/frontends/ecm_front_end_ipv4.c
+++ b/frontends/ecm_front_end_ipv4.c
@@ -144,7 +144,7 @@ static int ecm_front_end_ipv4_stop_handl
 	return ret;
 }
 
-static struct ctl_table ecm_front_end_ipv4_ctl_table[] = {
+static const struct ctl_table ecm_front_end_ipv4_ctl_table[] = {
 	{
 		.procname	= "front_end_ipv4_stop",
 		.data		= &ecm_front_end_ipv4_stopped,
--- a/frontends/ecm_front_end_ipv6.c
+++ b/frontends/ecm_front_end_ipv6.c
@@ -145,7 +145,7 @@ static int ecm_front_end_ipv6_stop_handl
 	return ret;
 }
 
-static struct ctl_table ecm_front_end_ipv6_ctl_table[] = {
+static const struct ctl_table ecm_front_end_ipv6_ctl_table[] = {
 	{
 		.procname	= "front_end_ipv6_stop",
 		.data		= &ecm_front_end_ipv6_stopped,
--- a/frontends/ppe/ecm_ppe_ipv4.c
+++ b/frontends/ppe/ecm_ppe_ipv4.c
@@ -962,7 +962,7 @@ static int ecm_ppe_ipv4_vlan_passthrough
 	return ret;
 }
 
-static struct ctl_table ecm_ppe_ipv4_ctl_table[] = {
+static const struct ctl_table ecm_ppe_ipv4_ctl_table[] = {
 	{
 		.procname	= "accel_limit_mode",
 		.data		= &ecm_ppe_ipv4_accel_limit_mode,
--- a/frontends/ppe/ecm_ppe_ipv6.c
+++ b/frontends/ppe/ecm_ppe_ipv6.c
@@ -1020,7 +1020,7 @@ static int ecm_ppe_ipv6_vlan_passthrough
 	return ret;
 }
 
-static struct ctl_table ecm_ppe_ipv6_ctl_table[] = {
+static const struct ctl_table ecm_ppe_ipv6_ctl_table[] = {
 	{
 		.procname	= "accel_limit_mode",
 		.data		= &ecm_ppe_ipv6_accel_limit_mode,
--- a/frontends/sfe/ecm_sfe_common.c
+++ b/frontends/sfe/ecm_sfe_common.c
@@ -497,7 +497,7 @@ bool ecm_sfe_ipv6_is_conn_limit_reached(
 
 #endif
 
-static struct ctl_table ecm_sfe_sysctl_tbl[] = {
+static const struct ctl_table ecm_sfe_sysctl_table[] = {
 	{
 		.procname	= "sfe_fast_xmit_enable",
 		.data		= &ecm_sfe_fast_xmit_enable,
@@ -537,7 +537,7 @@ static struct ctl_table ecm_sfe_sysctl_t
  */
 int ecm_sfe_sysctl_tbl_init(void)
 {
-	ecm_sfe_ctl_tbl_hdr = register_sysctl(ECM_FRONT_END_SYSCTL_PATH, ecm_sfe_sysctl_tbl);
+	ecm_sfe_ctl_tbl_hdr = register_sysctl(ECM_FRONT_END_SYSCTL_PATH, ecm_sfe_sysctl_table);
 	if (!ecm_sfe_ctl_tbl_hdr) {
 		DEBUG_WARN("Unable to register ecm_sfe_sysctl_tbl");
 		return -EINVAL;
--- a/frontends/sfe/ecm_sfe_ipv4.c
+++ b/frontends/sfe/ecm_sfe_ipv4.c
@@ -1105,7 +1105,7 @@ static int ecm_sfe_ipv4_reject_accelerat
 }
 #endif
 
-static struct ctl_table ecm_sfe_ipv4_ctl_table[] = {
+static const struct ctl_table ecm_sfe_ipv4_ctl_table[] = {
 	{
 		.procname	= "accel_limit_mode",
 		.data		= &ecm_sfe_ipv4_accel_limit_mode,
--- a/frontends/sfe/ecm_sfe_ipv6.c
+++ b/frontends/sfe/ecm_sfe_ipv6.c
@@ -1104,7 +1104,7 @@ static int ecm_sfe_ipv6_reject_accelerat
 }
 #endif
 
-static struct ctl_table ecm_sfe_ipv6_ctl_table[] = {
+static const struct ctl_table ecm_sfe_ipv6_ctl_table[] = {
 	{
 		.procname	= "accel_limit_mode",
 		.data		= &ecm_sfe_ipv6_accel_limit_mode,
--- a/frontends/sfe/ecm_sfe_multicast_ipv4.c
+++ b/frontends/sfe/ecm_sfe_multicast_ipv4.c
@@ -3219,7 +3219,7 @@ static int ecm_front_end_ipv4_mc_stop_ha
 	return ret;
 }
 
-static struct ctl_table ecm_sfe_multicast_ipv4_ctl_table[] = {
+static const struct ctl_table ecm_sfe_multicast_ipv4_ctl_table[] = {
 	{
 		.procname	= "ecm_sfe_multicast_ipv4_stop",
 		.data		= &ecm_front_end_ipv4_mc_stopped,
--- a/frontends/sfe/ecm_sfe_multicast_ipv6.c
+++ b/frontends/sfe/ecm_sfe_multicast_ipv6.c
@@ -3212,7 +3212,7 @@ static int ecm_front_end_ipv6_mc_stop_ha
 	return ret;
 }
 
-static struct ctl_table ecm_sfe_multicast_ipv6_ctl_table[] = {
+static const struct ctl_table ecm_sfe_multicast_ipv6_ctl_table[] = {
 	{
 		.procname	= "ecm_sfe_multicast_ipv6_stop",
 		.data		= &ecm_front_end_ipv6_mc_stopped,
