--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ KERNELVERSION := $(word 1, $(subst ., ,$
 SFE_BASE_OBJS := sfe.o sfe_init.o
 SFE_IPV4_OBJS := sfe_ipv4.o sfe_ipv4_udp.o sfe_ipv4_tcp.o sfe_ipv4_icmp.o sfe_ipv4_pppoe_br.o sfe_ipv4_multicast.o
 SFE_IPV6_OBJS := sfe_ipv6.o sfe_ipv6_udp.o sfe_ipv6_tcp.o sfe_ipv6_icmp.o sfe_ipv6_pppoe_br.o sfe_ipv6_multicast.o sfe_ipv6_etherip.o sfe_ipv6_frag.o
-SFE_PPPOE_OBJS := sfe_pppoe.o sfe_pppoe_mgr.o
+SFE_PPPOE_OBJS := sfe_pppoe.o
 ifndef SFE_256M_PROFILE
 ifndef SFE_16M_PROFILE
 SFE_IPV4_OBJS += sfe_ipv4_tun6rd.o sfe_ipv4_esp.o
@@ -71,4 +71,4 @@ ccflags-y += -DSFE_DSLITE_ENABLE
 endif
 endif
 
-ccflags-y += -Werror -Wall -Iexports/
+ccflags-y += -Wno-error -Wno-unused-function -Wno-unused-variable -Wno-missing-prototypes -Wall -I$(src)/exports/ -I$(src)/include/ -include $(src)/sfe_compat.h -DSFE_OPENWRT_COMPAT
