--- a/sfe_ipv4_udp.c
+++ b/sfe_ipv4_udp.c
@@ -169,8 +169,10 @@ int sfe_ipv4_recv_udp(struct sfe_ipv4 *s
 #ifdef SFE_SW_L4_CHECKSUM_VERIFICATION
 	struct net *net;
 #endif
+#ifndef SFE_OPENWRT_COMPAT
 	struct netdev_hw_offload_ops *offload_ops = NULL;
 	bool tun_vp_tx = false;
+#endif
 
 	/*
 	 * Is our packet too short to contain a valid UDP header?
@@ -773,7 +775,9 @@ int sfe_ipv4_recv_udp(struct sfe_ipv4 *s
 	features = cm->features;
 
 	fast_xmit = !!(cm->flags & SFE_IPV4_CONNECTION_MATCH_FLAG_FAST_XMIT);
+#ifndef SFE_OPENWRT_COMPAT
 	tun_vp_tx = !!(cm->flags & SFE_IPV4_CONNECTION_MATCH_FLAG_PPE_TUN_VP_VALID);
+#endif
 
 	/*
 	 * In the map-t tunnel, it needs correct transport header.
@@ -794,6 +798,7 @@ int sfe_ipv4_recv_udp(struct sfe_ipv4 *s
 	 * For tunnel Packets check if tunnel is active in PPE.
 	 * If tunnel is active queue the packet to tunnel VP
 	 */
+#ifndef SFE_OPENWRT_COMPAT
 	if (unlikely(tun_vp_tx)) {
 		offload_ops = rcu_dereference(xmit_dev->offload_ops);
 		if (likely(offload_ops && offload_ops->xmit)) {
@@ -805,6 +810,7 @@ int sfe_ipv4_recv_udp(struct sfe_ipv4 *s
 			}
 		}
 	}
+#endif
 
 	rcu_read_unlock();
 
--- a/sfe_ipv4_tcp.c
+++ b/sfe_ipv4_tcp.c
@@ -163,8 +163,10 @@ int sfe_ipv4_recv_tcp(struct sfe_ipv4 *s
 #ifdef SFE_SW_L4_CHECKSUM_VERIFICATION
 	struct net *net;
 #endif
+#ifndef SFE_OPENWRT_COMPAT
 	struct netdev_hw_offload_ops *offload_ops = NULL;
 	bool tun_vp_tx = false;
+#endif
 
 	/*
 	 * Is our packet too short to contain a valid TCP header?
@@ -933,7 +935,9 @@ int sfe_ipv4_recv_tcp(struct sfe_ipv4 *s
 	}
 	features = cm->features;
 	fast_xmit = !!(cm->flags & SFE_IPV4_CONNECTION_MATCH_FLAG_FAST_XMIT);
+#ifndef SFE_OPENWRT_COMPAT
 	tun_vp_tx = !!(cm->flags & SFE_IPV4_CONNECTION_MATCH_FLAG_PPE_TUN_VP_VALID);
+#endif
 
 	/*
 	 * In the map-t tunnel, it needs correct transport header.
@@ -947,6 +951,7 @@ int sfe_ipv4_recv_tcp(struct sfe_ipv4 *s
 	 * For tunnel Packets check if tunnel is active in PPE.
 	 * If tunnel is active queue the packet to tunnel VP
 	 */
+#ifndef SFE_OPENWRT_COMPAT
 	if (unlikely(tun_vp_tx)) {
 		offload_ops = rcu_dereference(xmit_dev->offload_ops);
 		if (likely(offload_ops && offload_ops->xmit)) {
@@ -958,6 +963,7 @@ int sfe_ipv4_recv_tcp(struct sfe_ipv4 *s
 			}
 		}
 	}
+#endif
 
 	rcu_read_unlock();
 
--- a/sfe_ipv6_udp.c
+++ b/sfe_ipv6_udp.c
@@ -154,8 +154,10 @@ int sfe_ipv6_recv_udp(struct sfe_ipv6 *s
 #ifdef SFE_SW_L4_CHECKSUM_VERIFICATION
 	struct net *net;
 #endif
+#ifndef SFE_OPENWRT_COMPAT
 	struct netdev_hw_offload_ops *offload_ops = NULL;
 	bool tun_vp_tx = false;
+#endif
 
 	DEBUG_TRACE("%px: sfe: sfe_ipv6_recv_udp called.\n", skb);
 
@@ -763,7 +765,9 @@ int sfe_ipv6_recv_udp(struct sfe_ipv6 *s
 	features = cm->features;
 
 	fast_xmit = !!(cm->flags & SFE_IPV6_CONNECTION_MATCH_FLAG_FAST_XMIT);
+#ifndef SFE_OPENWRT_COMPAT
 	tun_vp_tx = !!(cm->flags & SFE_IPV6_CONNECTION_MATCH_FLAG_PPE_TUN_VP_VALID);
+#endif
 
 	this_cpu_inc(si->stats_pcpu->packets_forwarded64);
 
@@ -771,6 +775,7 @@ int sfe_ipv6_recv_udp(struct sfe_ipv6 *s
 	 * For tunnel Packets check if tunnel is active in PPE.
 	 * If tunnel is active queue the packet to tunnel VP
 	 */
+#ifndef SFE_OPENWRT_COMPAT
 	if (unlikely(tun_vp_tx)) {
 		offload_ops = rcu_dereference(xmit_dev->offload_ops);
 		if (likely(offload_ops && offload_ops->xmit)) {
@@ -782,6 +787,7 @@ int sfe_ipv6_recv_udp(struct sfe_ipv6 *s
 			}
 		}
 	}
+#endif
 
 	rcu_read_unlock();
 
--- a/sfe_ipv6_tcp.c
+++ b/sfe_ipv6_tcp.c
@@ -132,8 +132,10 @@ int sfe_ipv6_recv_tcp(struct sfe_ipv6 *s
 #ifdef SFE_SW_L4_CHECKSUM_VERIFICATION
 	struct net *net;
 #endif
+#ifndef SFE_OPENWRT_COMPAT
 	struct netdev_hw_offload_ops *offload_ops = NULL;
 	bool tun_vp_tx = false;
+#endif
 
 	/*
 	 * Is our packet too short to contain a valid TCP header?
@@ -904,7 +906,9 @@ int sfe_ipv6_recv_tcp(struct sfe_ipv6 *s
 	features = cm->features;
 
 	fast_xmit = !!(cm->flags & SFE_IPV6_CONNECTION_MATCH_FLAG_FAST_XMIT);
+#ifndef SFE_OPENWRT_COMPAT
 	tun_vp_tx = !!(cm->flags & SFE_IPV6_CONNECTION_MATCH_FLAG_PPE_TUN_VP_VALID);
+#endif
 
 	this_cpu_inc(si->stats_pcpu->packets_forwarded64);
 
@@ -912,6 +916,7 @@ int sfe_ipv6_recv_tcp(struct sfe_ipv6 *s
 	 * For tunnel Packets check if tunnel is active in PPE.
 	 * If tunnel is active queue the packet to tunnel VP.
 	 */
+#ifndef SFE_OPENWRT_COMPAT
 	if (unlikely(tun_vp_tx)) {
 		offload_ops = rcu_dereference(xmit_dev->offload_ops);
 		if (likely(offload_ops && offload_ops->xmit)) {
@@ -923,6 +928,7 @@ int sfe_ipv6_recv_tcp(struct sfe_ipv6 *s
 			}
 		}
 	}
+#endif
 
 	rcu_read_unlock();
 
