--- a/sfe.c
+++ b/sfe.c
@@ -39,7 +39,7 @@
 
 extern int max_ipv4_conn;
 extern int max_ipv6_conn;
-extern bool fast_tc_filter;
+bool fast_tc_filter = false;
 uint8_t sfe_fls_flush_on_hwm_enable;
 
 #define SFE_MESSAGE_VERSION 0x1
@@ -2853,7 +2853,9 @@ int sfe_init_if(void)
 		goto exit2;
 	}
 
+#ifndef SFE_OPENWRT_COMPAT
 	sfe_pppoe_mgr_init();
+#endif
 
 	/*
 	 * Enable RFS by default
@@ -2892,7 +2894,9 @@ void sfe_exit_if(void)
 	 */
 	RCU_INIT_POINTER(athrs_fast_nat_recv, NULL);
 
+#ifndef SFE_OPENWRT_COMPAT
 	sfe_pppoe_mgr_exit();
+#endif
 
 	/*
 	 * Wait for all callbacks to complete.
--- a/sfe_ipv4.c
+++ b/sfe_ipv4.c
@@ -1513,9 +1513,11 @@ int sfe_ipv4_recv(struct net_device *dev
 	if (unlikely(sfe_l2_parse_flag_check(l2_info, SFE_L2_PARSE_FLAGS_PPPOE_INGRESS)) &&
 	    unlikely(sfe_pppoe_get_br_accel_mode() == SFE_PPPOE_BR_ACCEL_MODE_EN_3T)) {
 		struct ethhdr *eth = eth_hdr(skb);
+#ifndef SFE_OPENWRT_COMPAT
 		if (!sfe_pppoe_mgr_find_session(l2_info->pppoe_session_id, eth->h_source)) {
 			return sfe_ipv4_recv_pppoe_bridge(si, skb, dev, len, iph, ihl, l2_info);
 		}
+#endif
 	}
 
 	protocol = iph->protocol;
--- a/sfe_ipv6.c
+++ b/sfe_ipv6.c
@@ -1239,9 +1239,11 @@ int sfe_ipv6_recv(struct net_device *dev
 	if (unlikely(sfe_l2_parse_flag_check(l2_info, SFE_L2_PARSE_FLAGS_PPPOE_INGRESS)) &&
 	    unlikely(sfe_pppoe_get_br_accel_mode() == SFE_PPPOE_BR_ACCEL_MODE_EN_3T)) {
 		struct ethhdr *eth = eth_hdr(skb);
+#ifndef SFE_OPENWRT_COMPAT
 		if (!sfe_pppoe_mgr_find_session(l2_info->pppoe_session_id, eth->h_source)) {
 			return sfe_ipv6_recv_pppoe_bridge(si, skb, dev, len, iph, ihl, l2_info);
 		}
+#endif
 	}
 
 	if (IPPROTO_UDP == next_hdr) {
